### ISSUE TITLE
clang-static-analyzer false positive with transparent_union

### ISSUE THREAD

**Issue by fanf2** [llvm/llvm-project#60482]
The `ccc-analyzer` incorrectly reports a function argument is an uninitialized value (`core.CallAndMessage`) when the variable has been initialized by another function that was passed a pointer to the variable via a transparent union.

Example code:

```
#include <stdio.h>

struct a {
	void *ap;
};
struct b {
	void *bp;
};

typedef union buggy {
	struct a *a;
	struct b *b;
} buggy __attribute__((__transparent_union__));

static void
unpack(buggy u, struct a *a) {
	struct b *b = u.b;
	*b = (struct b){ .bp = a->ap };
}

extern void
invoke(struct a *a) {
	struct b b;
	unpack(&b, a);
	printf("a %p b.bp %p\n", a, b.bp);
}
```

Analyzer report:

```
$ ccc-analyzer -Wall -Wextra -c scan-build-bug.c
scan-build-bug.c:25:2: warning: 3rd function call argument is an uninitialized value [core.CallAndMessage]
        printf("a %p b.bp %p\n", a, b.bp);
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
```
